spec.html

```diff
@@ -36049,6 +36071,116 @@ <h1>
             1. Return _index_ + _cp_.[[CodeUnitCount]].
           </emu-alg>
         </emu-clause>
+
+        <emu-clause id="sec-getstringindex" type="abstract operation">
+          <h1>
+            GetStringIndex (
+              _S_: a String,
+              _e_: a non-negative integer,
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. If _S_ is the empty String, return 0.
+            1. Let _codepoints_ be StringToCodePoints(_S_).
+            1. Let _eUTF_ be the smallest index into _S_ that corresponds to the character at element _e_ of _codepoints_. If _e_ is greater than or equal to the number of elements in _codepoints_, then _eUTF_ is the number of code units in _S_.
+            1. Return _eUTF_.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-match-records">
+          <h1>Match Records</h1>
+          <p>A <dfn variants="Match Records">Match Record</dfn> is a Record value used to encapsulate the start and end indices of a regular expression match or capture.</p>
+          <p>Match Records have the fields listed in <emu-xref href="#table-match-record"></emu-xref>.</p>
+          <emu-table id="table-match-record" caption="Match Record Fields">
+            <table>
+              <tr>
+                <th>Field Name</th>
+                <th>Value</th>
+                <th>Meaning</th>
+              </tr>
+              <tr>
+                <td>[[StartIndex]]</td>
+                <td>a non-negative integer</td>
+                <td>The number of code units from the start of a string at which the match begins (inclusive).</td>
+              </tr>
+              <tr>
+                <td>[[EndIndex]]</td>
+                <td>an integer &ge; [[StartIndex]]</td>
+                <td>The number of code units from the start of a string at which the match ends (exclusive).</td>
+              </tr>
+            </table>
+          </emu-table>
+        </emu-clause>
+
+        <emu-clause id="sec-getmatchstring" type="abstract operation">
+          <h1>
+            GetMatchString (
+              _S_: a String,
+              _match_: a Match Record,
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Assert: _match_.[[StartIndex]] is an integer value &ge; 0 and &le; the length of _S_.
```

Comment:
```suggestion
            1. Assert: _match_.[[StartIndex]] is a non-negative integer value &le; the length of _S_.
```

I don't really like using &ge; and &le; in ways that are not binary infix. These usages are kind of unary-y and it's weird. This change will at least reduce it. We could also spell everything out with words, as in `less than or equal to the length of _S_`.